### DISCUSSION TITLE
How can the dual simplex iteration path be greatly changed by setting different parameters in "options.h"

### DISCUSSION THREAD

**Discussion by tusiqi1** [ERGO-Code/HiGHS#1324] [Q&A] [answered]
When solving the LP of the MIP relaxation for the first time, I want to use different dual simplex iteration strategies in parallel, and choose the best iteration strategy apply to all the subsequent LP solutions for the current MIP example.

I can change the iteration path of the dual simplex by using the parameters "simplex_dual_edge_weight_strategy" and "dual_simplex_cost_perturbation_multiplier", but I'm not sure that's a good option.

Hope professore could give me some advices~

**Comment by jajhall** [comment]
I don't think that there is any advantage to be gained this way

**Reply by tusiqi1** [reply]
In my tests, the overall geometric mean slowed down, although there were a few instances that sped up. Why is that?

**Reply by jajhall** [reply, marked as answer]
Although there's scope for parameter tuning on particular instance classes, the default parameters are well set for good general performance.

**Reply by tusiqi1** [reply]
Thanks !

**Reply by tusiqi1** [reply]
After tuning the parameters of LP in recent days, MIPLIB2017 geometric average speedup reached 4.9%. However, if only look at the examples (38) with non-default LP parameters selected, the geometric mean speedup is 18.5%.

**Reply by tusiqi1** [reply]
There are many MIP problems whose solving speed is independent of LP computing speed

**Comment by jajhall** [comment]
> After tuning the parameters of LP in recent days, MIPLIB2017 geometric average speedup reached 4.9%. However, if only look at the examples (38) with non-default LP parameters selected, the geometric mean speedup is 18.5%.

> There are many MIP problems whose solving speed is independent of LP computing speed

I can believe that you've managed to get this speed-up by modifying the LP parameters, but that the reason for the speed-up is that the MIP solver takes a different path that's more efficient. This ties in with your observation - that I know and have hear from others - that (within reason) MIP performance is independent of LP computing speed.

Are these non-default LP parameters problem-specific, or have you got one particular set that you've used for all MIPs

A good test of the validity of your tuning is to run through the same set of problems with the same LP option strategy, but with a different value for the random seed. If you've included the random seed in your problem-specific option settings, then significant speed-up is unsurprising.

**Reply by tusiqi1** [reply]
I got one particular set that used for all MIPs. 

The core idea is to assume that the faster the simplex iteration, the faster the MIP solution. In my experiments, I found that this assumption might be true if:

(1) When use the default paramaters, the LP solve time shoud be large. (such as longer than 1s)
(2) The newly set parameters need to perform much better than the default values. (such as 20% better.)

The large restriction resulted in only 38 test examples select non-default parameters, leading to a slight improvement.  If weake this limit, the speedup of MIPLIB may become negative
